pkg/signer/signer.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright The Rekor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package signer
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/sigstore/cosign/pkg/cosign/kms/gcp"
```

Comment:
We should probably try to avoid importing cosign into rekor here too, can we move the signer interface up into sigstore itself?

Reply:
Yup! I think the signer interface is already there but the kms/gcp specific code isn't. Just opened https://github.com/sigstore/sigstore/pull/41 to move it.

Reply:
cosign imports should be removed now!